engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbProcessState.java

```diff
@@ -134,6 +135,16 @@ public void putProcess(final long key, final ProcessRecord processRecord) {
         processRecord.getBpmnProcessIdBuffer(), processRecord.getChecksumBuffer());
   }
 
+  @Override
+  public void updateProcessState(
+      final long processDefinitionKey, final PersistedProcessState state) {
+    this.processDefinitionKey.wrapLong(processDefinitionKey);
+    final var process = processColumnFamily.get(this.processDefinitionKey);
+    process.setState(state);
+    processColumnFamily.upsert(this.processDefinitionKey, process);
```

Comment:
❌ I believe insertion would be an indication of corrupt data, let's use `update` instead

Reply:
Can I blame this on IntelliJ's autocomplete? 🤡